src/test/java/org/openmrs/contrib/qaframework/automation/ClinicalVisitSteps.java

```diff
@@ -224,18 +225,17 @@ public void systemDeletesCondition() {
 
 	@When("a user clicks on Attachments link from patient visits dashboard")
 	public void loadAttachmentsPage() {
-		visitsDashboardPage = (PatientVisitsDashboardPage) dashboardPage.goToRecentVisits();
-		attachmentsPage = (AttachmentsPage) dashboardPage.goToAttachmentsPage().waitForPage();
+		attachmentsPage = (AttachmentsPage) visitsDashboardPage.goToAttachmentsPage().waitForPage();
 	}
 
 	@Then("the system loads Attachments page")
 	public void systemloadsAttachmentsPage() {
-		assertTrue(textExists("Attachments"));
+		assertPage(attachmentsPage.waitForPage());
 	}
 
 	@When("a user attaches patient supporting file")
-	public void addSupportingFile() {
-		attachmentsPage.setFileUrl("/home/opensource/Documents/Form.pdf");
+	public void addSupportingFile() throws IOException {
+		attachmentsPage.setFileUrl(attachmentsPage.createRandomFile());
```

Comment:
why using `setFileUrl()` to pass the created randomFile?.  are you not supposed to create a method in the attachement page for the field the attachment field and then use it to pass attachment.createRandonFile() instead?

Reply:
@kdaud sure I created in the attachment page that creates a random file. This method returns the url to the file created, therefore the setFileUrl method in the attachment page rides on the url that is returned by createRandomFile method. However, I could also pass the createRandomFile() in setFileUrl() within the attachmentsPage such that I can just call setFileUrl() here in the steps definition. Would that make sense? I am thinking I may not need to create another method for the attachment field since setFileUrl handles that. @kdaud what do you think? I hope I understood your question.

Reply:
[setFileUrl()](https://github.com/openmrs/openmrs-contrib-qaframework/blob/da4bbde0eebcf62fcc0918efc84cd97356bc88d9/src/test/java/org/openmrs/contrib/qaframework/page/AttachmentsPage.java#L33) is used to pass a url for a hard coded path(if you one wants to attach a file residing on his/her machine) and I think it would be a good idea to create another method that handles createdRandonFile().

Reply:
@Irenyak1 see further thought [here](https://github.com/openmrs/openmrs-contrib-qaframework/pull/248#discussion_r763713801)